### ISSUE TITLE
There are errors in your xml file: out of memory

### ISSUE THREAD

**Issue by TianboJi** [buglabs/node-xml2json#151]
I have a 1.7GB xml file which is exported from FileMaker Pro. When I use this module to parse it, it returns 
>Error: There are errors in your xml file: out of memory

Is there any way to solve it? I already add the parameter "--max-old-space-size=4096" but it doesn't work

**Comment by aakashmalhotra** [comment]
This plugin does not stream the file so the whole file is loaded into memory. Hence, the error you have.
Solution: Stream your data

**Comment by c4milo** [comment]
@aakashmalhotra is right, even though we use node-expat which parses the XML document in streaming, we don't stream the resulting JSON structure out of the function.

**Comment by Matthbo** [comment]
Isn't it a good idea to add an option to make it a stream?

**Comment by c4milo** [comment]
@matthbo, yes, it would be useful with big documents. I just don’t currently have the bandwidth to do it :/. Although, I’m open to reviewing contributions!

**Comment by cangSDARM** [comment]
I'm handling a 8G xml file. It's have the same problem for me. Hope it support streamable API in 2021

**Comment by aakashmalhotra** [comment]
Hi @cangSDARM, 

you can use if you are dealing with huge files https://github.com/astro/node-expat

**Comment by cangSDARM** [comment]
@aakashmalhotra Thanks. I'll check that.
